lib/Target/ExportSystemC/Patterns/SystemCEmissionPatterns.cpp

```diff
@@ -177,6 +177,26 @@ struct SignalEmitter : OpEmissionPattern<SignalOp> {
     p << " " << op.getName() << ";\n";
   }
 };
+
+/// Emit a systemc.instance.decl operation.
+struct InstanceDeclEmitter : OpEmissionPattern<InstanceDeclOp> {
+  using OpEmissionPattern::OpEmissionPattern;
+
+  MatchResult matchInlinable(Value value) override {
+    if (value.getDefiningOp<InstanceDeclOp>())
+      return Precedence::VAR;
```

Comment:
Is this sufficient to make sure this never inclines?

Reply:
I'm not sure what 'inclines' would mean in this context. If you mean 'inlines' then the name of the instance (field/member name in C++) should always inline when it is referenced using the SSA value at some place.

Reply:
Yeah, inlines. I was thinking that this did something different. This behavior looks correct to me.